manifests/hitch.pp

```diff
@@ -0,0 +1,111 @@
+# @summary Installs Hitch the SSL Offloading Proxy of Varnish Enterprise
+#
+# @param package_name
+#   Define used package name
+# @param package_ensure
+#   Ensure package
+# @param service_ensure
+#   Ensure Service status
+# @param service_name
+#   Service name for hitch (must match installed)
+# @param config_path
+#   Path for hitch config
+# @param config_template
+#   Used EPP Config template
+# @param frontends
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param backend
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param pem_files
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param ssl_engine
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param tls_protos
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param ciphers
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param ciphersuites
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param workers
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param backlog
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param keepalive
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param chroot
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param user
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param group
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param log_level
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param syslog
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param syslog_facility
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param daemon
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param write_proxy
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param sni_nomatch_abort
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param tcp_fastopen
+#   see Parameter in https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
+# @param alpn_protos
```

Comment:
can you maybe briefly explain what these parameters do instead of linking to the docs? you can add that as a dedicated puppet-strings element:

```
# @see https://github.com/varnish/hitch/blob/master/hitch.conf.man.rst
```

Reply:
@bastelfreak Is ist a good idea to dupplicate the documtentation of the vendor? These parameters are only set as written by them.